src/Jobs/GenerateCSVJob.php

```diff
@@ -131,12 +158,38 @@ public function setIncludeHeader($includeHeader)
         $this->IncludeHeader = $includeHeader;
     }
 
+    protected function makeDir($path)
+    {
+        // whether to use 'chmod' to override 'mkdir' perms which obey umask
+        $ignore_umask = $this->config()->get('ignore_umask');
+
+        // perms mode given to 'mkdir' and 'chmod'
+        $permission_mode = $this->config()->get('permission_mode');
+
+        // check mode is numeric string
+        if (is_string($permission_mode) && is_numeric($permission_mode)) {
+            // convert from octal to decimal for mkdir
+            $permission_mode = octdec($permission_mode);
+        } else {
+            // disallow non-string types as we can't ensure they are valid
+            throw new Exception("Only string values are allowed for 'permission_mode'");
```

Comment:
Minor suggestion - you could flip the condition and throw this exception if it's not a string or numeric, then you don't need an else

Reply:
Done, also moved all logic inside the `is_dir` check.